build.gradle

```diff
@@ -93,123 +91,125 @@ class PackageProperties {
-    !file(jreExtractDir).exists()
-}
+static boolean checkHash(JreProperties props) {
+    final hash = MessageDigest.getInstance('SHA-256').with {
+        props.jre.eachByte(1024) {buff,len ->
+            update(buff, 0, len)
+        }
+        digest().encodeHex() as String
+    }
 
-task createPackages(description: 'Create three packages for Linux, Windows and Mac', dependsOn:unzipJre) {
-    //This task also depends on other three tasks which are created by iterating osNames
+    hash == props.hash
 }
 
-packagePropertiesMap.each {String sys, PackageProperties props ->
-    props.with {
-        String scriptDir = "${projectDir}/launcher-scripts/${scriptFile}"
-        String jreExtractSubDir = "${jreExtractDir}/$sys"
-        Class taskCLass = Class.forName("org.gradle.api.tasks.bundling.${taskType}")
-
-        task "packageFor$sys"(type: taskCLass, description: "Create ${sys} package") {
-            mustRunAfter unzipJre
-            archiveBaseName = "admin-console-package-for-${sys}"
-            extension = packageFileExt
-            if(fileCompression) {
-                compression = Compression.GZIP
+task unzipJre(description: 'Unzip a JRE when the extract directory is missing', dependsOn: downloadJre) {
+    doFirst {
+
+        jreProperties.each {sys, props ->
+            // If any one of three JRE folders is missing then only unzip the corresponding JRE
+            if (!new File("${jreDownloadDir.absolutePath}/jre/${sys}").exists()) {
+                if (props.jre.name.endsWith(ZIP)) {
+                    copy {
+                        from zipTree(props.jre)
+                        into "$jreExtractDir/${WINDOWS}"
+                    }
+                } else {
+                    copy {
+                        from tarTree(resources.gzip(props.jre))
+                        into "$jreExtractDir/${props.jre.name.contains(MAC) ? MAC : LINUX}"
+                    }
+                }
             }
-            destinationDirectory = file(packageDir)
+        }
+    }
+}
+
+build.dependsOn(unzipJre)
 
-            from file(jarDir)
-            from file(jreExtractSubDir)
-            from file(scriptDir)
-            exclude (["*.${ZIP}", "*.${TARGZ}"])
+distributions {
+    packageProperties.each { String sys, PackageProperties props ->
+        String scriptDir = "${projectDir}/launcher-scripts/${props.scriptFile}"
+        String jreExtractSubDir = "${jreExtractDir}/$sys"
 
-            doLast {
-                println("Package for ${sys} is created under ${packageDir}.")
+        "${sys}Packages" {
+            baseName = "admin-console-package-for-${sys}"
+            contents {
+                from {file(jarDir)}
+                from {file(jreExtractSubDir)}
+                from {file(scriptDir)}
             }
         }
-
-        createPackages.dependsOn("packageFor$sys")
     }
 }
 
-build.finalizedBy(createPackages)
+// We want tar files are compressed by GZIP and their extension name is 'tar.gz'
```

Comment:
I think we need to reword:

    // We want TAR files to be compressed with GZIP with the file extension of '.tar.gz'